File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, arr: str) -> int:
        if len(arr)==0:
            return 0
        d=[-1]*256
        l,r=0,0
        res=0
        while r<len(arr):
            if d[ord(arr[r])]!=-1:
                if d[ord(arr[r])]>=l:
                    l=d[ord(arr[r])]+1
            d[ord(arr[r])]=r
            r+=1
            res=max(res,r-l+1)
        return res-1

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int>mpp(256,-1);
        int l=0;
        int r=0;
        int res=0;
        while (r<s.size()){
            if (mpp[s[r]]!=-1){
                if (mpp[s[r]]>=l){
                    l=mpp[s[r]]+1;
                }

            }
            mpp[s[r]]=r;
            res=max(res,r-l+1);
            r++;
        }
        return res;
        
    }
};